src/modvault/modmodel.py

```diff
@@ -0,0 +1,186 @@
+from PyQt5.QtCore import QAbstractListModel, Qt, QSortFilterProxyModel, QModelIndex
+from .modmodelitem import ModModelItem
+from enum import Enum
+
+
+class ModModel(QAbstractListModel):
```

Comment:
Same here, a lil bit of duplication. Guess it's a bit harder to fix here since add / remove signals have different names in different set classes - we'll have to plan out well how to deduplicate the stuff in the future.